vowpalwabbit/core/src/reductions/cb/cb_explore_adf_large_action_space.cc

```diff
@@ -72,6 +101,53 @@ void cb_explore_adf_large_action_space::calculate_shrink_factor(const ACTION_SCO
 
 inline void just_add_weights(float& p, float, float fw) { p += fw; }
 
+inline void no_op(float&, float, float) {}
+
+bool cb_explore_adf_large_action_space::generate_A(const multi_ex& examples)
+{
+  // TODO extend wildspace interactions before calling foreach
+  uint64_t row_index = 0;
+  uint64_t max_non_zero_col = 0;
+  // TODO check triplets type
+  _triplets.clear();
+  for (auto* ex : examples)
+  {
+    assert(!CB::ec_is_example_header(*ex));
+
+    float no_op_float = 0.f;
+    if (_all->weights.sparse)
+    {
+      triplet_constructor<sparse_parameters> w(_all->weights.sparse_weights, row_index, _triplets, max_non_zero_col);
+      GD::foreach_feature<float, float, no_op, triplet_constructor<sparse_parameters>>(w, _all->ignore_some_linear,
```

Comment:
Rather than hiding the construction logic in the `operator[]` it would make more sense to do the insertion in the function that is called per weight. This would also migrate more cleanly to if this function where to be able to use a lambda instead of context + fptr

Reply:
yeah done